### ISSUE TITLE
Union type is always generated, even if withObjectType is false

### ISSUE THREAD

**Issue by tobiasdiez** [Code-Hex/graphql-codegen-typescript-validation-schema#252]
If you have a schema 
```graphql
        type Square {
          size: Int
        }
        type Circle {
          radius: Int
        }
        union Shape = Circle | Square
```
then (with the default `withObjectType = false`) the generated code contains 
```ts
export function ShapeSchema() {
    return z.union([CircleSchema(), SquareSchema()])
}
```
but the referenced type schemas `CircleSchema` and `SquareSchema` don't exist. Thus, the union schema should also not be generated if `withObjectType` is false.

**Comment by sschneider-ihre-pvs** [comment]
have a similar problem with just using `union` as type alias

**Comment by sschneider-ihre-pvs** [comment]
@Code-Hex also the uniontypedefinition seems to be missing in the latest version?!

**Comment by sschneider-ihre-pvs** [comment]
![image](https://user-images.githubusercontent.com/86342787/204276265-a73a8ebe-f487-4e02-ba5d-6973a093ae05.png)

**Comment by Code-Hex** [comment]
@tobiasdiez Thanks I will fix it!

@sschneider-ihre-pvs I cannot read your JavaScript

**Comment by sschneider-ihre-pvs** [comment]
> @tobiasdiez Thanks I will fix it!
> 
> @sschneider-ihre-pvs I cannot read your JavaScript

that is the published js from the npm package

**Comment by anhdd-kuro** [comment]
@alisabzevari 
thought you can find the problem in #237

**Comment by Code-Hex** [comment]
@tobiasdiez @anhdd-kuro I just published the latest version (v0.7.1) which is contained this fix.

I'm sorry too late.

https://github.com/Code-Hex/graphql-codegen-typescript-validation-schema/releases/tag/v0.7.1

**Comment by tobiasdiez** [comment]
I can confirm that it is working now. Thanks for the fix!
